### PR TITLE
[DP-99 & 100] A Layout component for CWKW

### PR DESCRIPTION
**Pull request by patelShas**
A page has been created showing off what the CWKW layout looks like on the url [site]/cwkw-placeholder.

image

A new menu was created for this layout, and the menu components have also been changed to accept menuID's rather than being hard-coded for a specific one. So, the menu component still works for the main page:

image

Creating the cwkw layout also required making the preferences and mobile menu buttons change colors so they would remain visible. To show them, here is the site on mobile:

image

### CODE REVIEW

File: website/src/cwkw/cwkw-layout.css.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { style } from "@vanilla-extract/css"
+import * as baseLayout from "../layout.css"
+import { colors } from "./theme.css"
+
+export const header = style([baseLayout.header])
```
Comment (loafofpiecrust) [inline website/src/cwkw/cwkw-layout.css.ts:5]: If the style is just reusing another one, I think you can just do `export const header = baseLayout.header`. You could reduce this even further by doing `export { header, headerContents, siteTitle, siteLink } from "../layout.css"`

File: website/src/graphql/wordpress/queries.graphql [hunk 2]
```diff
@@ -34,3 +34,23 @@ query MainMenu {
     }
   }
 }
+
+query MenuByID($slug: Int!) {
+  menus(where: { id: $slug }) {
+    nodes {
+      databaseId
+      menuItems(where: { parentDatabaseId: 0 }) {
```
Comment (loafofpiecrust) [inline website/src/graphql/wordpress/queries.graphql:42]: Hmm why are you using `parentDatabaseId: 0` here? Also, I would rather us have one query definition for both menus instead of a separate one for the main site menu.

File: website/src/menu.css.ts [hunk 3]
```diff
@@ -53,7 +61,7 @@ export const drawerItem = style([
     display: "inline-block",
     padding: vspace.quarter,
     textDecoration: "none",
-    color: colors.text,
+    color: "black",
```
Comment (loafofpiecrust) [inline website/src/menu.css.ts:64]: Why this change? I'd rather add another color to the theme that handles this instead of using a hard coded color.

File: website/src/menu.css.ts [hunk 4]
```diff
@@ -45,6 +46,13 @@ export const navLink = style([
   },
 ])
 
+export const subMenuItems = style([
+  navLink,
+  {
+    color: "black",
```
Comment (loafofpiecrust) [inline website/src/menu.css.ts:52]: Use a theme color, I'd like to avoid hard-coded colors where possible.

File: website/src/menu.css.ts [hunk 5]
```diff
@@ -10,6 +10,7 @@ import sprinkles, {
   vspace,
 } from "src/sprinkles.css"
 import { paddingX, paddingY } from "src/style-utils"
+import * as cwkwTheme from "./cwkw/theme.css"
```
Comment (loafofpiecrust) [inline website/src/menu.css.ts:13]: Never use themes directly, always use the variables from the theme contract defined in the `sprinkles.css` file. That way if you switch themes, the styles will update accordingly.

File: website/src/menu.css.ts [hunk 6]
```diff
@@ -73,6 +81,11 @@ export const navButton = style([
   },
 ])
 
+export const navButtonChoices = styleVariants({
+  cwkw: [navButton, { color: cwkwTheme.colors.text }],
```
Comment (loafofpiecrust) [inline website/src/menu.css.ts:85]: So here, I wouldn't make variants. Instead change the `navButton` styles so that switching themes does this for us.

File: website/src/menu.tsx [hunk 7]
```diff
@@ -16,20 +16,28 @@ import {
...
   navLink,
   navMenu,
+  subMenuItems,
 } from "./menu.css"
 import { closeBlock } from "./sprinkles.css"
 
-export const NavMenu = () => {
+export const NavMenu = (p: { menuID: number }) => {
   const location = useLocation()
-  const [{ data }] = Wordpress.useMainMenuQuery()
```
Comment (loafofpiecrust) [inline website/src/menu.tsx:27]: Since you swapped out the main menu query, can we just get rid of it then?

File: website/src/menu.tsx [hunk 8]
```diff
@@ -16,20 +16,28 @@ import {
...
-  const menuItems = data?.menuItems?.nodes
+  const [{ data }] = Wordpress.useMenuByIdQuery({
+    variables: { slug: p.menuID },
+  })
+  const menus = data?.menus?.nodes
```
Comment (loafofpiecrust) [inline website/src/menu.tsx:32]: Why did you need to query `menus` instead of `menuItems`?

File: website/src/menu.tsx [hunk 9]
```diff
@@ -100,11 +108,18 @@ const SubMenu = ({ item, location }: { location: Location; item: any }) => {
   )
 }
 
-export const MobileNav = () => {
+export const MobileNav = (p: { menuID: number; cwkw?: boolean }) => {
```
Comment (loafofpiecrust) [inline website/src/menu.tsx:111]: Yeah so get rid of this `cwkw` prop and make all necessary changes through the theme.

File: website/src/mode.css.ts [hunk 10]
```diff
@@ -46,7 +47,10 @@ export const prefButtonShell = style({
   flex: 1,
 })
 
-export const prefButton = collPanelButton
+export const prefButton = styleVariants({
+  cwkw: [collPanelButton, { color: cwkwTheme.colors.text }],
+  base: [collPanelButton],
+})
```
Comment (loafofpiecrust) [inline website/src/mode.css.ts:53]: Same thing here, don't make a cwkw variant, use the theme for this change.

File: website/src/pages/documents/document.page.tsx [hunk 11]
```diff
@@ -10,6 +10,7 @@ import { DocumentAudio } from "src/audio-player"
 import { Breadcrumbs } from "src/breadcrumbs"
 import { Button } from "src/components"
 import Link from "src/components/link"
+import CWKWLayout from "src/cwkw/cwkw-layout"
```
Comment (loafofpiecrust) [inline website/src/pages/documents/document.page.tsx:13]: This looks like an unnecessary import

File: website/src/menu.tsx [hunk 12]
```diff
@@ -16,20 +16,28 @@ import {
...
 } from "./menu.css"
 import { closeBlock } from "./sprinkles.css"
 
-export const NavMenu = () => {
+export const NavMenu = (p: { menuID: number }) => {
   const location = useLocation()
-  const [{ data }] = Wordpress.useMainMenuQuery()
-  const menuItems = data?.menuItems?.nodes
+  const [{ data }] = Wordpress.useMenuByIdQuery({
+    variables: { slug: p.menuID },
```
Comment (loafofpiecrust) [inline website/src/menu.tsx:30]: I think we can call the query variable `id` instead of `slug` because its not a slug right?

File: website/src/menu.tsx (hunk 8 above)
Comment (patelShas) [reply website/src/menu.tsx:32]: Querying menuItems seemed to just return items without regard to which menu they came from, and their parentID was usually null and so couldn't be filtered by that. Mean while querying menus made it easy to see all the nodes of a specific wordpress menu by filtering by ID. So, it seemed easier to query menus than menuItems.

File: website/src/menu.css.ts (hunk 3 above)
Comment (patelShas) [reply website/src/menu.css.ts:64]: There was an odd quirk that made it so menu items were not visible against the background, which has been fixed.

File: website/src/graphql/wordpress/queries.graphql (hunk 2 above)
Comment (patelShas) [reply website/src/graphql/wordpress/queries.graphql:27]: I don't know why but including it makes the graphql query return Contact-Us, but excluding it means Contact-Us does not appear. I can't tell why, as the parentDatabaseID and parentID themselves are not different from other nodes. So, including this (where: { parentDatabaseId: 0 }) was the simplest way to bring back Contact-Us.

**Comment by patelShas** [comment]
Made most changes as well as some changes to the cwkw theme so that elements would be more visible.

image

File: website/src/cwkw/theme.css.ts [hunk 13]
```diff
@@ -28,4 +27,10 @@ export const themeClass = createTheme(theme, {
...
     smallCaps: "Charis SIL",
   },
-})
\ No newline at end of file
+})
+
+export const colors = {
+  ...theme.colors,
+  transparent: "transparent",
+  inherit: "inherit",
```
Comment (loafofpiecrust) [inline website/src/cwkw/theme.css.ts:35]: You don't need to define these in the colors object because they'll never change. When I said use only theme colors, I should've said "unless they're special values". For example we'll never need to change what `inherit` means but we may want to change what "the text color" means.

File: website/src/menu.css.ts (hunk 4 above)
Comment (loafofpiecrust) [reply website/src/menu.css.ts:51]: This still needs to be using the `colors` object.

File: website/src/pages/cwkw-placeholder.page.tsx [hunk 14]
```diff
@@ -0,0 +1,16 @@
+import CWKWLayout from "src/cwkw/cwkw-layout"
+
+const CWKWPlaceholderPage = () => (
+  <CWKWLayout>
+    <main>
```
Comment (loafofpiecrust) [inline website/src/pages/cwkw-placeholder.page.tsx:5]: Could you add here:

```tsx
<Helmet>
  <meta name=”robots” content=”noindex,nofollow”/>
</Helmet>
```

Helmet comes from `react-helmet`, you can look for other usages to see examples. This will prevent the page from being indexed since we don't want anyone to get linked to it.

File: website/src/sprinkles.css.ts [hunk 15]
```diff
@@ -41,7 +41,7 @@ export const themeClass = createTheme(theme, {
   },
   colors: {
     header: "#f7eeed",
-    button: "#f7eeed",
+    button: "black",
```
Comment (loafofpiecrust) [inline website/src/sprinkles.css.ts:44]: Why are you changing a color in the base theme?

File: website/src/cwkw/theme.css.ts (hunk 13 above)
Comment (patelShas) [reply website/src/cwkw/theme.css.ts:35]: Yeah, I've realized since that this whole css declaration of themes is unnecessary and I just misunderstood how the themes work when adding it in.

File: website/src/sprinkles.css.ts (hunk 15 above)
Comment (patelShas) [reply website/src/sprinkles.css.ts:44]: Beforehand the button was the same color as the header, so making buttons and links have the button color meant they were invisible in the base theme. So to fix this, I tried changing this to black. I see though this doesn't work well with other elements that already used the button color.

**Review by loafofpiecrust — APPROVED** [review]
